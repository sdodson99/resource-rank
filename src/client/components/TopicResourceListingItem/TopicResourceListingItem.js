import React from 'react';
import PropTypes from 'prop-types';
import styles from './TopicResourceListingItem.module.css';
import Link from 'next/link';
import ListingItem from '../ListingItem/ListingItem';
import RatingStars from '../RatingStars/rating-stars';
import ArrowForwardIcon from '@material-ui/icons/ArrowForward';

const TopicResourceListingItem = ({
  topicId,
  resourceId,
  name,
  link,
  rating,
}) => {
  const topicResourceLink = `/topics/${topicId}/resources/${resourceId}`;

  return (
    <div
      className={styles.TopicResourceListingItem}
      data-testid="TopicResourceListingItem"
    >
      <Link href={topicResourceLink}>
        <a>
          <ListingItem hover={true}>
            <div className="sm:flex sm:justify-between items-center sm:text-left text-center py-5 px-3">
              <div className="col-sm-auto flex flex-col items-center sm:items-start">
                <div>{name}</div>
                <div className="mt-1">
                  <RatingStars rating={rating} />
                </div>
              </div>
              <div className="mt-5 sm:mt-0">
                <ArrowForwardIcon />
              </div>
            </div>
          </ListingItem>
        </a>
      </Link>
    </div>
  );
};

TopicResourceListingItem.propTypes = {
  topicId: PropTypes.string,
  resourceId: PropTypes.string,
  name: PropTypes.string,
  link: PropTypes.string,
  rating: PropTypes.number,
};

TopicResourceListingItem.defaultProps = {};

export default TopicResourceListingItem;
