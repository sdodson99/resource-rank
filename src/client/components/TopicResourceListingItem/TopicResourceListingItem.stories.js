import React from 'react';
import TopicResourceListingItem from './TopicResourceListingItem';

export default {
  component: TopicResourceListingItem,
  title: 'Components/TopicResourceListingItem',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <TopicResourceListingItem {...args} />;

export const Default = Template.bind({});
Default.args = {
  topicId: '1',
  resourceId: '1',
  topicSlug: 'topic-1',
  resourceSlug: 'resource-1',
  name: 'Resource 1',
  rating: 5,
};

export const Verified = Template.bind({});
Verified.args = {
  ...Default.args,
  verified: true,
};
