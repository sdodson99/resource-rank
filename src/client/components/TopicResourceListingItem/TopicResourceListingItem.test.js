import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import TopicResourceListingItem from './TopicResourceListingItem';

describe('<TopicResourceListingItem />', () => {
  test('it should mount', () => {
    render(<TopicResourceListingItem />);

    const topicResourceListingItem = screen.getByTestId(
      'TopicResourceListingItem'
    );

    expect(topicResourceListingItem).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(
        <TopicResourceListingItem
          topicId={'123'}
          topicSlug={'topic-name'}
          resourceSlug={'resource-name'}
          rating={3}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when verified', () => {
    const tree = renderer
      .create(
        <TopicResourceListingItem
          topicId={'123'}
          topicSlug={'topic-name'}
          resourceSlug={'resource-name'}
          rating={3}
          verified={true}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
