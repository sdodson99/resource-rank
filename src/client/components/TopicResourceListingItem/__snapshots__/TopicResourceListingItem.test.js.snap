// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicResourceListingItem /> should render correctly 1`] = `
<div
  className="TopicResourceListingItem"
  data-testid="TopicResourceListingItem"
>
  <a
    href="/topics/123/resources/123"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="ListingItem HoverListingItem"
      data-testid="ListingItem"
    >
      <div
        className="sm:flex sm:justify-between items-center sm:text-left text-center py-5 px-3"
      >
        <div
          className="col-sm-auto flex flex-col items-center sm:items-start"
        >
          <div />
          <div
            className="mt-1"
          >
            <div
              className="flex"
            >
              <div
                className="me-1"
                onClick={[Function]}
                onMouseOver={[Function]}
              >
                <svg
                  aria-label="star-empty"
                  fill="#d6d6d6"
                  height={20}
                  viewBox="0 0 16 16"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                  />
                </svg>
              </div>
              <div
                className="me-1"
                onClick={[Function]}
                onMouseOver={[Function]}
              >
                <svg
                  aria-label="star-empty"
                  fill="#d6d6d6"
                  height={20}
                  viewBox="0 0 16 16"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                  />
                </svg>
              </div>
              <div
                className="me-1"
                onClick={[Function]}
                onMouseOver={[Function]}
              >
                <svg
                  aria-label="star-empty"
                  fill="#d6d6d6"
                  height={20}
                  viewBox="0 0 16 16"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                  />
                </svg>
              </div>
              <div
                className="me-1"
                onClick={[Function]}
                onMouseOver={[Function]}
              >
                <svg
                  aria-label="star-empty"
                  fill="#d6d6d6"
                  height={20}
                  viewBox="0 0 16 16"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                  />
                </svg>
              </div>
              <div
                className="me-1"
                onClick={[Function]}
                onMouseOver={[Function]}
              >
                <svg
                  aria-label="star-empty"
                  fill="#d6d6d6"
                  height={20}
                  viewBox="0 0 16 16"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="mt-5 sm:mt-0"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </div>
      </div>
    </div>
  </a>
</div>
`;
